File: protocol/CsProtocol/Packet/ObjectB.cs
using System;
using System.Collections.Generic;
using CsProtocol.Buffer;

namespace CsProtocol
{
    
    public class ObjectB : IProtocol
    {
        public bool flag;

        public static ObjectB ValueOf(bool flag)
        {
            var packet = new ObjectB();
            packet.flag = flag;
            return packet;
        }


        public short ProtocolId()
        {
            return 103;
        }
    }


    public class ObjectBRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 103;
        }

        public void Write(ByteBuffer buffer, IProtocol packet)
        {
            if (buffer.WritePacketFlag(packet))
            {
                return;
            }
            ObjectB message = (ObjectB) packet;
            buffer.WriteBool(message.flag);
        }

        public IProtocol Read(ByteBuffer buffer)
        {
            if (!buffer.ReadBool())
            {
                return null;
            }
            ObjectB packet = new ObjectB();
            bool result0 = buffer.ReadBool();
            packet.flag = result0;
            return packet;
        }
    }
}

File: protocol/zfoopy/packet/ObjectB.py

class ObjectB:

    flag = False  # bool

    def protocolId(self):
        return 103

    @classmethod
    def write(cls, buffer, packet):
        if packet is None:
            buffer.writeInt(0)
            return
        buffer.writeInt(-1)
        buffer.writeBool(packet.flag)
        pass

    @classmethod
    def read(cls, buffer):
        length = buffer.readInt()
        if length == 0:
            return None
        beforeReadIndex = buffer.getReadOffset()
        packet = ObjectB()
        result0 = buffer.readBool() 
        packet.flag = result0
        if length > 0:
            buffer.setReadOffset(beforeReadIndex + length)
        return packet

